awx/main/scheduler/task_manager.py

```diff
@@ -432,6 +434,12 @@ def generate_dependencies(self, undeped_tasks):
                     continue
                 latest_inventory_update = self.get_latest_inventory_update(inventory_source)
                 if self.should_update_inventory_source(task, latest_inventory_update):
+                    if inventory_source.source == "scm":
+                        latest_src_project_update = self.get_latest_project_update(inventory_source.source_project_id)
+                        if self.should_update_related_project(task, latest_src_project_update):
```

Comment:
This could still need more work to avoid spawning the same update twice when the playbook comes from the same project. There's also an optimization issue here, because if the inventory update was just spawned, you would already have the update in memory, but this would re-fetch it.